### DISCUSSION TITLE
High detail only around POIs

### DISCUSSION THREAD

**Discussion by mikaelhg** [onthegomap/planetiler#775] [Q&A] [answered]
I'd like to produce a small ~30MB pmtiles file that contains z0-z9 tiles for the whole county of Finland, and z10-z14 tiles that are a few km in distance from a dozen of POIs. Is this possible with mere configuration?

My fallback would be to produce a normal z0-z9 pmtiles file, and a second file for z10-14 for polygons around my POIs. I assume this can be done with a polygon file where you include multiple areas in one .poly file, one for each POI interest bound?

The use case is where the user needs to see detail around the POI, practically how to navigate around the POI, but also on the same map see where the POI is in relation to cities and roads, available at z9. 

A whole country z0-z14 pmfiles file is around 800MB, which is too large for some of the tools in the workflow.

**Comment by cmahnke** [comment]
That would make a great feature!
How do you envisage it to work with the POIs? By feeding in an list of bounding boxes or using points and a radius for example?

**Reply by mikaelhg** [reply]
A YAML configuration file that doesn't need to re-define everything in the standard configuration, but can either refer to or include the standard configuration, and modify it for areas in a referenced .poly file?

**Reply by msbarry** [reply]
The underlying [TileExtents](https://github.com/onthegomap/planetiler/blob/main/planetiler-core/src/main/java/com/onthegomap/planetiler/geo/TileExtents.java) utility that handles bounding boxes supports an arbitrary x/y -> boolean function per zoom-level so you could construct what you want in code, it's just a matter of what's the right API to provide this via config at runtime...?

**Reply by cmahnke** [reply]
Another idea to configure this would be a simple GeoJSON file:
```
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "geometry": {
          "type": "Polygon", 
          "coordinates": [
              [
                  [30.0, 10.0],
                  [40.0, 40.0],
                  [20.0, 40.0],
                  [10.0, 20.0],
                  [30.0, 10.0]
              ]
          ]
      },
      "properties": {
        "maxzoom": "10"
      }
    },
    {
      "type": "Feature",
      "geometry": {
        "type": "Point",
        "coordinates": [102.0, 0.5]
      },
      "properties": {
        "radius": "10",
        "maxzoom": 15
      }
    }
  ]
}
```
Note: Coordinates are made up

Semantics of handling points needs to be defined, there could be a `radius` or `bbox` property. Other arguments like `maxzoom`could also be passed by property.

**Reply by msbarry** [reply, marked as answer]
I like that approach, or we could also have a way to list bounding boxes or poly/geojson files with a max zoom like `--bounds=world<=3;10,20,30,40<=10;pois.geojson<15` to have full planet tiles up to z3, tiles within bounding box 10,20,30,40 up to z10, and around pois.geojson up to z14 ?  That would also resolve #133

I would say if a maxzoom is missing, then max maxzoom (z14/15) would be implied.
